src/datamodule.py

```diff
@@ -10,17 +12,42 @@
 
 
 # %%
-def _array_to_torch(filepath: str) -> torch.Tensor:
+def _array_to_torch(filepath: str) -> dict[str, torch.Tensor | str]:
     """
-    Read a GeoTIFF file using rasterio into a numpy.ndarray, and convert it
-    to a torch.Tensor (float16 dtype).
+    Read a GeoTIFF file using rasterio into a numpy.ndarray, convert it to a
+    torch.Tensor (float16 dtype), and also output spatiotemporal metadata
+    associated with the image.
+
+    Parameters
+    ----------
+    filepath : str
+        The path to the GeoTIFF file.
+
+    Returns
+    -------
+    outputs : dict
+        A dictionary containing the following items:
+        - image: torch.Tensor - multi-band raster image with shape (Band, Height, Width)
+        - bbox: torch.Tensor - spatial bounding box as (xmin, ymin, xmax, ymax)
+        - crs: torch.Tensor - coordinate reference system as an EPSG code
+        - date: str - the date the image was acquired in YYYY-MM-DD format
     """
     # GeoTIFF - Rasterio
     with rasterio.open(fp=filepath) as dataset:
+        # Get image data
         array: np.ndarray = dataset.read()
         tensor: torch.Tensor = torch.as_tensor(data=array.astype(dtype="float16"))
 
-    return tensor
+        # Get spatial bounding box and coordinate reference system in UTM projection
+        bbox: torch.Tensor = torch.as_tensor(  # xmin, ymin, xmax, ymax
+            data=dataset.bounds, dtype=torch.float64
+        )
+        crs: int = torch.as_tensor(data=dataset.crs.to_epsg(), dtype=torch.int32)
+
+        # Get date
+        date: str = pathlib.Path(filepath).name[15:25]  # YYYY-MM-DD format
+
+    return {"image": tensor, "bbox": bbox, "crs": crs, "date": date}
```

Comment:
Tempted to store this in an Arrow Table with [`FixedShapedTensorArray`](https://arrow.apache.org/docs/python/generated/pyarrow.FixedShapeTensorArray.html) for the image and bbox 'columns'. Possibly revisit this in the future.